apps/explorer/app/contract/[id]/page.tsx

```diff
@@ -42,44 +38,58 @@ export default async function Page({ params }) {
     ),
   ])
 
-  if (error) {
-    throw error
-  }
+  if (rateError) throw rateError
+  if (!rate) return notFound()
```

Comment:
This should probably throw an error so the error page is rendered rather than 404.

Reply:
We didn't talk about this one yesterday. I just changed it to throw, but would you mind explaining your logic about why throw versus the other "successful" requests where we check for undefined and `return notFound()`? Trying to see if there's something here I need to know!

Reply:
notFound/404 should be for when the user navigates to a contract ID that does not exist, so if we were to fetch the contract data and it 400s (not found) rather than 500s (server error). 
In this case, if the supporting rate data errors its definitely not a "contract not found", its a server error.

Reply:
Okay, so does that mean that all of the return notFound's that are not specifically the contract call should be throws instead?

Reply:
yeah I would first check the contract error, if its a 400 then return notFound, if the contract error is a 500 or any other required data fetching errors that can all be thrown to the error page.